app/views/contests/scoreboard.html.erb

```diff
@@ -9,17 +9,27 @@
   //-->
   </script>
 <% end %>
+<% show_scoreboard = @contest.live_scoreboard || @contest.ended? || policy(@contest).inspect? %>
 <% if !@contest.started? %>
   <em>The contest has not started yet.</em>
+<% elsif !show_scoreboard && (!user_signed_in? || !@contest.has_competitor?(current_user)) %>
```

Comment:
Would it be clearer if the negation was factored out?
```suggestion
<% elsif !(show_scoreboard || (user_signed_in? && @contest.has_competitor?(current_user))) %>
```
I think this makes it slightly clearer that the `else` part (which displays the scoreboard) will be executed if show_scoreboard is true or the user is competing.